### PR TITLE
MINIFI-311 Move to alpine base for docker image.

### PR DESCRIPTION
**Pull request by achristianson**
This change takes the 1.5G ubuntu minifi image and reduces it to a 39MB image by using a multi-stage alpine build.

MINIFI-311 ported dockerfile to alpine
MINIFI-311 allow warnings on Linux for compatibility with Alpine
MINIFI-311 fixed spdlog path in BuildTests cmake file
MINIFI-349 moved to multi-stage docker build

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: thirdparty/spdlog-20170710/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Adapted from various sources, including:
```
Comment (apiri) [inline thirdparty/spdlog-20170710/.travis.yml]: We should remove this from the source inclusion.  Anything beyond the actual source should get stripped out to minimize LICENSE/NOTICE concerns.

File: thirdparty/spdlog-20170710/bench/boost-bench-mt.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
+//
```
Comment (apiri) [inline thirdparty/spdlog-20170710/bench/boost-bench-mt.cpp]: Should remove the entirety of the bench subdirectory.

File: thirdparty/spdlog-20170710/example/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# *************************************************************************/
```
Comment (apiri) [inline thirdparty/spdlog-20170710/example/CMakeLists.txt]: Should remove the entirety of the example directory

File: thirdparty/spdlog-20170710/tests/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#
```
Comment (apiri) [inline thirdparty/spdlog-20170710/tests/CMakeLists.txt]: Should remove the tests subdirectory and skip this part of the build for the lib

**Comment by achristianson** [comment]
Thanks for the feedback. I will make the required changes and update the PR.

**Comment by achristianson** [comment]
@apiri removed those dirs & rebased. Should be good to go now.

**Comment by apiri** [comment]
reviewing changes

**Comment by apiri** [comment]
hey @achristianson,

everything looks good here. will get this merged in.  thanks for your adjustments!

**Pull request closed without merging**